### ISSUE TITLE
RPM postuninst script fails to run

### ISSUE THREAD

**Issue by ximon18** [NLnetLabs/routinator#773]
Platform: Fedora 36 x86_64

```
$ ls -la /usr/share/routinator/rpm/
total 12
drwxr-xr-x 1 root root   54 Aug 23 22:03 .
drwxr-xr-x 1 root root    6 Aug 23 22:02 ..
-rwxr-xr-x 1 root root 1073 Aug 23 17:53 postinst
-rwxr-xr-x 1 root root  378 Aug 23 17:53 postuninst
-rwxr-xr-x 1 root root  436 Aug 23 17:53 preuninst
```

Clearly the `postuninst` script exists.

```
$ sudo dnf remove routinator
Dependencies resolved.
==============================================================================================================================================================================================================================================
 Package                                                    Architecture                                           Version                                                      Repository                                               Size
==============================================================================================================================================================================================================================================
Removing:
 routinator                                                 x86_64                                                 0.12.0-dev-1                                                 @System                                                 8.6 M

Transaction Summary
==============================================================================================================================================================================================================================================
Remove  1 Package

Freed space: 8.6 M
Is this ok [y/N]: y
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                                                                      1/1 
  Running scriptlet: routinator-0.12.0-dev-1.x86_64                                                                                                                                                                                       1/1 
  Erasing          : routinator-0.12.0-dev-1.x86_64                                                                                                                                                                                       1/1 
  Running scriptlet: routinator-0.12.0-dev-1.x86_64                                                                                                                                                                                       1/1 
/var/tmp/rpm-tmp.Q0i4Rw: line 1: /usr/share/routinator/rpm/postuninst: No such file or directory
warning: %postun(routinator-0:0.12.0-dev-1.x86_64) scriptlet failed, exit status 127

Error in POSTUN scriptlet in rpm package routinator
  Verifying        : routinator-0.12.0-dev-1.x86_64                                                                                                                                                                                       1/1 

Removed:
  routinator-0.12.0-dev-1.x86_64                                                                                                                                                                                                              

Complete!
```

Notice these errors:
```
/var/tmp/rpm-tmp.Q0i4Rw: line 1: /usr/share/routinator/rpm/postuninst: No such file or directory
warning: %postun(routinator-0:0.12.0-dev-1.x86_64) scriptlet failed, exit status 127

Error in POSTUN scriptlet in rpm package routinator
```

**Comment by ximon18** [comment]
So we realized that the workaround of specifying a command that runs a script file doesn't work for `post_uninstall_script` as by the time it is run the script has been removed from disk.

Still not wanting to pollute my `Cargo.toml` with Bash scripts I'm now trying to see if the new `--set-metadata` functionality can help here, with:

```
cargo generate-rpm --set-metadata "post_uninstall_script = \"\"\"$(cat pkg/rpm/postuninst)\"\"\""
```

This looks fine when the created package is inspected:

```
$ rpm -qp --scripts target/generate-rpm/routinator-0.12.0-dev-1.x86_64.rpm 
postinstall scriptlet:
/usr/share/routinator/rpm/postinst $*
preuninstall scriptlet:
/usr/share/routinator/rpm/preuninst $*
postuninstall scriptlet:
#!/bin/bash -e
# Script based on the RPM %systemd_postun scriptlet. See:
#   - https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_systemd
#   - https://cgit.freedesktop.org/systemd/systemd/tree/src/core/macros.systemd.in

systemctl daemon-reload >/dev/null 2>&1 || :
if [ $1 -ge 1 ] ; then
    systemctl try-restart routinator.service >/dev/null 2>&1 || :
fi
```

**Comment by ximon18** [comment]
Fixed by PR #780.
